File: typescript/src/abstract-factory/concrete-factory1.ts
import type { AbstractFactory } from "./abstract-factory.ts";
import type { AbstractProductA } from "./abstract-product-a.ts";
import type { AbstractProductB } from "./abstract-product-b.ts";
import { ProductA1 } from "./product-a1.ts";
import { ProductB1 } from "./product-b1.ts";

/** Creates the products of family 1. */
export class ConcreteFactory1 implements AbstractFactory {
  createProductA(): AbstractProductA {
    return new ProductA1();
  }

  createProductB(): AbstractProductB {
    return new ProductB1();
  }
}

File: python/src/patterns/abstractfactory/concrete_factory1.py
from __future__ import annotations

from typing import TYPE_CHECKING

from .product_a1 import ProductA1
from .product_b1 import ProductB1

if TYPE_CHECKING:
    from .abstract_product_a import AbstractProductA
    from .abstract_product_b import AbstractProductB


class ConcreteFactory1:
    """Creates the products of family 1."""

    def create_product_a(self) -> AbstractProductA:
        return ProductA1()

    def create_product_b(self) -> AbstractProductB:
        return ProductB1()
